delta/extensions/losses.py

```diff
@@ -28,11 +28,39 @@
 from delta.config import config
 from delta.config.extensions import register_loss
 
+
+def suggest_filter_size(image1, image2, power_factors, filter_size):
+    '''Figure out if we need to shrink the filter to accomodate a smaller
+       input image'''
+
+    cap = 2**(len(power_factors)-1)
+    if not(image1.shape[0]/cap >= filter_size and
+           image1.shape[1]/cap >= filter_size and
+           image1.shape[0]/cap >= filter_size and
+           image2.shape[1]/cap >= filter_size):
+        H = tf.math.reduce_min((image1.shape, image2.shape))
+        suggested_filter_size = int(H/(2**(len(power_factors)-1)))
+    else:
+        suggested_filter_size = filter_size
+    return suggested_filter_size
+
 def ms_ssim(y_true, y_pred):
     """
-    `tf.image.ssim_multiscale` as a loss function.
-    """
-    return 1.0 - tf.image.ssim_multiscale(y_true, y_pred, 4.0)
+    `tf.image.ssim_multiscale` as a loss function. This loss function requires two
+    dimensional inputs.
+    """
+    y_true = tf.cond(tf.math.less(tf.rank(y_true), 3),
```

Comment:
What is this for?

Reply:
If the input is [x, y] it needs to be expanded to [x, y, 1].